docs/standard/microservices-architecture/multi-container-microservice-net-applications/background-tasks-with-ihostedservice.md

```diff
@@ -0,0 +1,260 @@
+---
+title: Implement background tasks in microservices with IHostedService and the BackgroundService class
+description: .NET Microservices Architecture for Containerized .NET Applications | Implement background tasks in microservices with IHostedService and the BackgroundService class
+keywords: Docker, Microservices, ASP.NET, Container
+author: CESARDELATORRE
+ms.author: wiwagn
+ms.date: 12/07/2017
+ms.prod: .net-core
+ms.technology: dotnet-docker
+ms.topic: article
+---
+# Implement background tasks in microservices with IHostedService and the BackgroundService class
+
+Background tasks and scheduled jobs are something you might need to implement, eventually, in a microservice based application or in any kind of application. The difference when using a microservices architecture is that you can implement a single microservice process/container for hosting these background tasks so you can scale it down/up as you need or you can even make sure that it runs a single instance of that microservice process/container.
+
+From a generic point of view, in .NET Core we called these type of tasks as Hosted Services, because they are services/logic that you host within your host/application/microservice. Note that in this case, the hosted service simply means a class with the background task logic.
+
+Since .NET Core 2.0, the framework provides a new interface named IHostedService helping you to easily implement hosted services. The basic idea is that you can register multiple background tasks (hosted services), that run in the background while your web host or host is running, as shown in the image below.
```

Comment:
`IHostedService` should be in code styling 

Or even better, an xref: `<xref:Microsoft.Extensions.Hosting.IHostedService>`  That automatically generates a link to the API reference.

Reply:
OK, so I'm putting <xref:Microsoft.Extensions.Hosting.IHostedService> instead of IHostedService, for the first reference. Afterwards, just as code styling.